src/shared/components/dropdown/dropdown.component.ts

```diff
@@ -11,31 +12,38 @@ import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
       provide: NG_VALUE_ACCESSOR,
       useExisting: DropdownComponent,
       multi: true,
-    }
+    },
   ],
 })
 export class DropdownComponent implements ControlValueAccessor {
-  statusOptions: string[] = Object.keys(Status);
+  @Input()
+  value: Status | SortTerm;
+
+  @Input()
+  name: IOption['value'];
+
+  @Input()
+  statusClass: Status;
 
   @Input()
-  status: Status;
+  icon: string;
+
+  @Input()
+  options: IOption[];
 
   private onChange: (value: string) => void = () => {};
   private onTouched: () => void = () => {};
 
-  get statusClass(): Status {
-    return Status[this.status];
-  }
-
-  setStatus(status: string): void {
-    if (this.status !== status) {
-      this.writeValue(status);
+  setStatus(value: Status | SortTerm, translatedValue: string): void {
+    if (this.value !== value) {
+      this.writeValue(value);
     }
+    this.name = translatedValue;
   }
```

Comment:
your displayed text should be set by getting an option from the options array by selected optionId